pandas/core/indexes/base.py

```diff
@@ -3627,7 +3627,7 @@ def _reindex_non_unique(self, target):
             else:
 
                 # need to retake to have the same size as the indexer
-                indexer[~check] = 0
+                indexer[~check] = -1
```

Comment:
what effect did this have? e.g. was this a bug before or just not used?

Reply:
To the best of my understanding, it was not used: not in the sense that this line was not hit, but because the locations for missing keys were then taken from ``new_indexer`` (see three lines below), where they were correctly marked with -1.